app/src/androidTest/java/org/simple/clinic/overdue/AppointmentRepositoryAndroidTest.kt

```diff
@@ -2402,6 +2406,171 @@ class AppointmentRepositoryAndroidTest {
     assertThat(overdueAppointmentsCount).isEqualTo(2)
   }
 
+  @Test
+  fun fetching_overdue_appointments_should_work_correctly() {
+    fun createOverdueAppointment(
+        patientUuid: UUID,
+        scheduledDate: LocalDate,
+        facilityUuid: UUID,
+        patientAssignedFacilityUuid: UUID?
+    ) {
+      val patientProfile = TestData.patientProfile(
+          patientUuid = patientUuid,
+          generatePhoneNumber = true,
+          patientAssignedFacilityId = patientAssignedFacilityUuid
+      )
+      patientRepository.save(listOf(patientProfile)).blockingAwait()
+
+      val bp = TestData.bloodPressureMeasurement(
+          patientUuid = patientUuid,
+          facilityUuid = facilityUuid
+      )
+      bpRepository.save(listOf(bp)).blockingAwait()
+
+      val bloodSugar = TestData.bloodSugarMeasurement(
+          patientUuid = patientUuid,
+          facilityUuid = facilityUuid
+      )
+      bloodSugarRepository.save(listOf(bloodSugar)).blockingAwait()
+
+      val appointment = TestData.appointment(
+          patientUuid = patientUuid,
+          facilityUuid = facilityUuid,
+          scheduledDate = scheduledDate,
+          status = Scheduled,
+          cancelReason = null
+      )
+      appointmentRepository.save(listOf(appointment)).blockingAwait()
+    }
+
+    //given
+    val patientWithOneDayOverdue = UUID.fromString("1e5f206b-2bc0-4e5e-bbbe-4f4a1bdaca53")
+    val patientWithTenDaysOverdue = UUID.fromString("50604030-303a-4979-bfda-297c169ed929")
+    val patientWithFifteenDaysOverdue = UUID.fromString("a4e5c0e0-cacd-49bf-8663-ab4d19435c3b")
+    val patientWithOverAnYearDaysOverdue = UUID.fromString("9e1c9dae-6bea-4463-8ad8-609d9118e20d")
+
+    val now = LocalDate.now(clock)
+    val facility1Uuid = UUID.fromString("4f3c6c64-3a2b-4f18-9179-978c2aa0b698")
+    val facility2Uuid = UUID.fromString("ca52615f-402d-48f1-a063-4d6af19baaa6")
+
+    val assignedFacilityUuid = facility1Uuid
+
+    val facility1 = TestData.facility(uuid = facility1Uuid, name = "PHC Obvious")
+    val facility2 = TestData.facility(uuid = facility2Uuid, name = "PHC Bagta")
+
+    facilityRepository.save(listOf(facility1, facility2)).blockingAwait()
+
+    createOverdueAppointment(
+        patientUuid = patientWithOneDayOverdue,
+        scheduledDate = now.minusDays(1),
+        facilityUuid = facility1Uuid,
+        patientAssignedFacilityUuid = assignedFacilityUuid
+    )
+    createOverdueAppointment(
+        patientUuid = patientWithTenDaysOverdue,
+        scheduledDate = now.minusDays(10),
+        facilityUuid = facility1Uuid,
+        patientAssignedFacilityUuid = null
+    )
+    createOverdueAppointment(
+        patientUuid = patientWithFifteenDaysOverdue,
+        scheduledDate = now.minusDays(15),
+        facilityUuid = facility2Uuid,
+        patientAssignedFacilityUuid = assignedFacilityUuid
+    )
+    createOverdueAppointment(
```

Comment:
Can we also verify the following cases?

- A patient with a different assigned facility and appointment in `facility1` is *NOT* showing up.
- A patient with no assigned facility and appointment in `faciltiy1` which is less than a year overdue shows up.

Reply:
Done